backend/inspirehep/assign/tasks.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import structlog
+from celery import shared_task
+from inspire_schemas.builders import LiteratureBuilder
+from invenio_db import db
+from invenio_pidstore.errors import PIDDoesNotExistError
+from jsonschema import ValidationError
+from sqlalchemy.exc import (
+    DisconnectionError,
+    OperationalError,
+    ResourceClosedError,
+    TimeoutError,
+    UnboundExecutionError,
+)
+from sqlalchemy.orm.exc import NoResultFound, StaleDataError
+
+from inspirehep.records.api import ConferencesRecord, LiteratureRecord
+from inspirehep.records.errors import MissingArgumentError
+
+LOGGER = structlog.getLogger()
+
+
+@shared_task(
+    ignore_results=False,
+    queue="assign",
+    acks_late=True,
+    retry_backoff=2,
+    retry_kwargs={"max_retries": 6},
+    autoretry_for=(
+        NoResultFound,
+        StaleDataError,
+        DisconnectionError,
+        TimeoutError,
+        UnboundExecutionError,
+        ResourceClosedError,
+        OperationalError,
+    ),
+)
+def assign_paper_to_conference(literature_recids, conference_recid):
+    try:
+        conference = ConferencesRecord.get_record_by_pid_value(conference_recid)
+    except PIDDoesNotExistError:
+        LOGGER.exception(
+            "Cannot assign papers to conference. Conference does not exist.",
+            conference_recid=conference_recid,
+        )
+    conference_ref = conference.get("self")
+    cnum = conference.get("cnum")
```

Comment:
Aren't they always present?

Reply:
https://github.com/inspirehep/inspire-schemas/blob/afc2553e150fc83e72aa3187926caa126a0a096d/inspire_schemas/records/conferences.yml#L98-L100 https://github.com/inspirehep/inspire-schemas/blob/afc2553e150fc83e72aa3187926caa126a0a096d/inspire_schemas/records/conferences.yml#L265-L266
They are not mandatory. It's possible that some old conference is missing those.